### PR TITLE
Update DataSource LastTransitionTime when populated PVC is updated

### PR DESCRIPTION
**Pull request by arnongilboa**
Signed-off-by: Arnon Gilboa <agilboa@redhat.com>

**What this PR does / why we need it**:
The `DataSourceReady` condition `LastTransitionTime` does not show a timestamp of the last source PVC update. It only shows a timestamp for its initial population, although the PVC name is updated.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
Update DataSource LastTransitionTime when populated source PVC is updated
```

### CODE REVIEW

**Comment by awels** [comment]
So it looks like there is no good way of detecting that the current PVC has changed, which in turn leads us to doing what you did. Would it make sense to add a `currentSourcePVC` that contains the `DataVolumeSourcePVC` of the current source. Then if the spec changes, we can detect this change because the status doesn't match the spec, we can then clear out the ready condition and everything else takes care of itself.

**Comment by arnongilboa** [comment]
> So it looks like there is no good way of detecting that the current PVC has changed, which in turn leads us to doing what you did. Would it make sense to add a `currentSourcePVC` that contains the `DataVolumeSourcePVC` of the current source. Then if the spec changes, we can detect this change because the status doesn't match the spec, we can then clear out the ready condition and everything else takes care of itself.

so why not add a mutating webhook to reset the Ready condition whenever source PVC is updated?

**Comment by mhenriks** [comment]
I think a new status field is a better solution.  If the resource never transitioned from not ready to ready, the transition time should not change.

**Comment by arnongilboa** [comment]
> I think a new status field is a better solution. If the resource never transitioned from not ready to ready, the transition time should not change.

added

File: staging/src/kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1/types.go [hunk 1]
```diff
@@ -443,6 +443,8 @@ type DataSourceSource struct {
 
 // DataSourceStatus provides the most recently observed status of the DataSource
 type DataSourceStatus struct {
+	// Source is the current source of the data referenced by the DataSource
+	Source     DataSourceSource      `json:"source"`
```
Comment (akrejcir) [inline staging/src/kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1/types.go:447]: Making this field required in the JSON, breaks API backward compatibility. Can you consider marking it as `omitempty`?

Comment (arnongilboa) [reply staging/src/kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1/types.go:447]: Sure, sorry I missed it.

Comment (arnongilboa) [reply staging/src/kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1/types.go:447]: @akrejcir https://github.com/kubevirt/containerized-data-importer/pull/2382

**Pull request merged**